openmdao/core/problem.py

```diff
@@ -2285,67 +2325,101 @@ def _assemble_deriv_data(params, resids, cdata, jac_fwd, jac_rev, jac_fd,
             if out_stream is None:
                 continue
 
-            if started:
-                out_stream.write(' -'*30 + '\n')
+            if compact_print:
+                if jac_fwd and jac_rev:
+                    if not started:
+                        out_str_tmpl = "{0} wrt {1} | {2} | {3} |  {4} | {5} | {6} | {7} | {8}\n"
+                        out_str = out_str_tmpl.format(_pad_name('<unknown>'), _pad_name('<param>'),
+                            _pad_name('fwd mag.', 10, quotes=False), _pad_name('rev mag.', 10, quotes=False),
+                            _pad_name('fd mag.', 10, quotes=False), _pad_name('a(fwd-fd)', 10, quotes=False),
+                            _pad_name('a(rev-fd)', 10, quotes=False), _pad_name('r(fwd-rev)', 10, quotes=False),
+                            _pad_name('r(rev-fd)', 10, quotes=False)
+                        )
+                        out_stream.write(out_str)
+                        out_stream.write('-'*len(out_str)+'\n')
+                        started=True
+
+                    out_str_tmpl = "{0} wrt {1} | {2:.4e} | {3:.4e} |  {4:.4e} | {5:.4e} | {6:.4e} | {6:.4e} | {6:.4e}\n"
```

Comment:
there is a bug here. `{6:.4e} | {6:.4e} | {6:.4e}` should be `{6:.4e} | {7:.4e} | {8:.4e}`